### PR TITLE
Add new command: plugins refresh

### PR DESCRIPTION
**Pull request by denCalderon**
Add new `hekit plugins` subcommand `plugins refresh`.
This command synchronizes the metadata of the plugins TOML file with the metadata of all plugins present on the system.

### CODE REVIEW

File: kit/utils/subparsers.py [hunk 1]
```diff
@@ -56,8 +56,46 @@ def discover_subparsers_plugins(module_dirs: List[str], kit_root: PathType):
         yield from funcs
 
 
-def discover_subparsers_kit(module_dirs: List[str], kit_root: PathType):
-    """Import modules in module_dirs, and discover and return a generator of set_.*_subparser functions"""
+def get_plugin_arg_choices(
+    plugin_name: str,
+    plugin_root: PathType = PluginsConfig.ROOT_DIR,
+) -> List[str]:
+    """Return the choices of the argument parser"""
```
Comment (jlhcrawford) [inline kit/utils/subparsers.py:63]:
```suggestion
    """Return the choices (list of plugin names) of the argument parser"""
```

File: kit/commands/plugin.py [hunk 2]
```diff
@@ -167,47 +167,107 @@ def move_plugin_data(
...
+    """Remove a plugin folder if exists"""
+    plugin_path = dest_path / plugin_name
+    if Path(plugin_path).exists():
+        rmtree(plugin_path)
+
+
+def are_plugin_args_correct(
+    plugin_name: str, is_plugin_present: bool, hekit_parsers_list: List[str]
+) -> bool:
+    """Verify that trser_choices_he plugin argument is unique"""
```
[the remark this answers is not in the document]
Comment (jlhcrawford) [reply kit/commands/plugin.py:182]: Maybe change this to something like "Verify that all plugin names are unique"

File: kit/commands/plugin.py [hunk 3]
```diff
@@ -167,47 +167,107 @@ def move_plugin_data(
...
+    arg_choices_list = get_plugin_arg_choices(plugin_name)
+
+    # verify it contains only one choice
```
Comment (jlhcrawford) [inline kit/commands/plugin.py:185]:
```suggestion
    # verify the plugin name only appears once
```

**Review by jlhcrawford — CHANGES_REQUESTED** [review]
Implementation looks fine. Mainly documentation/clarity of implementation.

File: kit/PLUGINS.md [hunk 4]
```diff
@@ -86,6 +88,13 @@ In order to enable a plugin again, run the command
 hekit plugins enable plugin-name
 ```
 
+#### refresh
+Execute the following command to synchronize the information in plugin
```
Comment (jlhcrawford) [inline kit/PLUGINS.md:92]:
```suggestion
Execute the following command to synchronize the information in the plugin
```

File: tests/test_cmd_plugins.py [hunk 5]
```diff
@@ -211,6 +211,10 @@ def test_move_plugin_data_zip(tmp_path):
     assert (dest_path / plugin_name).exists()
 
 
+def test_are_plugin_args_correct():
```
[the remark this answers is not in the document]
Comment (denCalderon) [reply tests/test_cmd_plugins.py:214]: this is already resolved in the lastest version

File: tests/test_cmd_plugins.py [hunk 6]
```diff
@@ -443,22 +446,35 @@ def test_list_plugins_enable(mocker):
...
-    mockers = Mockers(mocker)
+    mockers = Mockers(mocker, plugin_name)
 
     list_plugins(args)
     assert 2 == mockers.mock_print.call_count
-    mockers.mock_print.assert_called_with("plugin2    1.0.0      disabled  ")
+    mockers.mock_print.assert_called_with("plugin2    2.3.0      disabled  ")
+
+
+def test_refresh_plugins():
```
[the remark this answers is not in the document]
Comment (denCalderon) [reply tests/test_cmd_plugins.py:459]: this is already resolved in the lastest version

File: kit/utils/constants.py [hunk 7]
```diff
@@ -26,6 +26,22 @@ class Constants:  # pylint: disable=too-many-instance-attributes
     # Root directory
     HEKIT_ROOT_DIR: Path = Path(__file__).resolve().parent.parent.parent
 
+    # HE comands
```
Comment (hamishun) [inline kit/utils/constants.py:29]:
```suggestion
    # hekit core commands
```

File: kit/utils/constants.py [hunk 8]
```diff
@@ -26,6 +26,22 @@ class Constants:  # pylint: disable=too-many-instance-attributes
...
+        "build",
+        "fetch",
+        "list",
+        "new",
+        "plugins",
+        "remove",
+        "algebras",
+        "gen-primes",
+    ]
+
```
Comment (hamishun) [inline kit/utils/constants.py:44]: Maybe a set would be better?
```suggestion
    HEKIT_COMMANDS = {
        "check-dependencies",
        "docker-build",
        "init",
        "install",
        "build",
        "fetch",
        "list",
        "new",
        "plugins",
        "remove",
        "algebras",
        "gen-primes",
    }

```

**Review by hamishun — CHANGES_REQUESTED** [review]

File: kit/utils/constants.py (hunk 7 above)
Comment (denCalderon) [reply kit/utils/constants.py]: updated

File: kit/utils/constants.py (hunk 8 above)
Comment (denCalderon) [reply kit/utils/constants.py]: updated

File: tests/test_cmd_plugins.py [hunk 9]
```diff
@@ -211,6 +213,78 @@ def test_move_plugin_data_zip(tmp_path):
...
+    act_value = are_plugin_args_correct(plugin_name, False, {plugin_name})
+    assert act_value == False
+    mockers.mock_print.assert_called_with(
+        f"Invalid argument definition, '{argument_name}' is already used by another plugin"
+    )
+
+
+def test_are_plugin_args_correct_unique(mocker):
+    """Verify that the SW does not report an error when
+    the argument name is fine"""
```
[the remark this answers is not in the document]
Comment (denCalderon) [reply tests/test_cmd_plugins.py]: updated

File: tests/test_cmd_plugins.py [hunk 10]
```diff
@@ -211,6 +213,78 @@ def test_move_plugin_data_zip(tmp_path):
     assert (dest_path / plugin_name).exists()
 
 
+def test_are_plugin_args_correct_elements(mocker):
+    """Verify that the SW reports an error when
+    there is not argument name in the list"""
```
[the remark this answers is not in the document]
Comment (denCalderon) [reply tests/test_cmd_plugins.py]: updated

File: kit/utils/constants.py [hunk 11]
```diff
@@ -26,6 +26,22 @@ class Constants:  # pylint: disable=too-many-instance-attributes
     # Root directory
     HEKIT_ROOT_DIR: Path = Path(__file__).resolve().parent.parent.parent
 
+    # Hekit core commands
```
Comment (jlhcrawford) [inline kit/utils/constants.py:29]:
```suggestion
    # hekit core commands
```

**Review by jlhcrawford — CHANGES_REQUESTED** [review]

**Review by hamishun — APPROVED** [review]

**Review by jlhcrawford — APPROVED** [review]

**Pull request merged**